tee-worker/litentry/core/evm-dynamic-assertions/src/repository.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020-2024 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+use crate::{
+	sealing::io::{seal_state, unseal_state},
+	AssertionId, SmartContractByteCode,
+};
+use lc_dynamic_assertion::AssertionLogicRepository;
+use std::{
+	collections::HashMap,
+	format,
+	string::{String, ToString},
+	vec::Vec,
+};
+
+#[cfg(feature = "sgx")]
+use std::sync::SgxMutex as Mutex;
+
+#[cfg(feature = "std")]
+use std::sync::Mutex;
+
+pub type AssertionsMap = HashMap<AssertionId, (SmartContractByteCode, Vec<String>)>;
+
+// Assertion repository backed by sealed file. Contains only latest state from parachain storage.
+pub struct EvmAssertionRepository {
+	path: String,
+	state: Mutex<AssertionsMap>,
+}
+
+impl EvmAssertionRepository {
+	pub fn new(path: &str) -> Result<Self, String> {
+		let mut state = HashMap::new();
+		let unsealed = unseal_state(path)
+			.map_err(|e| format!("Could not unseal assertions state: {:?}", e))?;
+		unsealed.into_iter().for_each(|(id, byte_code)| {
+			state.insert(id, byte_code);
+		});
+
+		Ok(EvmAssertionRepository { state: state.into(), path: path.to_string() })
+	}
+}
+
+impl AssertionLogicRepository for EvmAssertionRepository {
+	type Id = AssertionId;
+	type Item = (SmartContractByteCode, Vec<String>);
+
+	fn get(&self, id: &Self::Id) -> Result<Option<Self::Item>, String> {
```

Comment:
How would this work in multi worker setup? I think the sealed files will get sync during initialization only (please correct me if I'm wrong)

Reply:
This is the initial sync, after that assertions are synced with parachain state in `event_handler.rs`

Reply:
True, I forgot about that 👍🏼